Source/Utils/Source/Logging/Log.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2024 Steven Frost and Orion Lyau
+
+#include <Utils/Logging/Log.h>
+
+#include <chrono>
+#include <cstdarg>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+// -----------------------------------------------------------------------------
+
+namespace Utils
+{
+namespace Logging
+{
+
+// -----------------------------------------------------------------------------
+
+std::string GetTimestampString()
+{
+	const auto CurrentTime = std::chrono::system_clock::now();
+	const auto TimeSinceEpoch = std::chrono::system_clock::to_time_t( CurrentTime );
+	const auto* TimeComponents = gmtime( &TimeSinceEpoch );
+
+	std::stringstream StringStream;
+	StringStream << std::put_time( TimeComponents, "%FT%T" );
+
+	const auto TruncatedTime = std::chrono::system_clock::from_time_t( TimeSinceEpoch );
```

Comment:
Can we rename `TimeSinceEpoch` to something like `SecondsSinceEpoch`?  It didn't seem obvious to me what was happening initially until I realized that converting to `time_t` was doing it in whole seconds,

Also I wasn't sure about `TruncatedTime` either until I figured out the context, but I'm not sure what a better name might be.  Maybe it'd be obvious when it's created from `SecondsSinceEpoch`?